agent/php_user_instrument_llist.h

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if LOOKUP_METHOD == LOOKUP_USE_LINKED_LIST
+
+/*
+ * Purpose : Store zend_function's metadata (filename, lineno) in wraprec
+ *           for use in looking up the instrumentation.
+ * 
+ * Params  : 1. The wraprec to update with metadata
+ *           2. The zend function with meta data
+ *
+ * Returns : void
+ */
+static void wraprec_metadata_set(nruserfn_t* wraprec, zend_function* func) {
+    const char* filename = nr_php_function_filename(func);
+  /*
+   * Before setting a filename, ensure it is not NULL and it doesn't equal the
+   * "special" designation given to funcs without filenames. If the function is
+   * an evaluated expression or called directly from the CLI there is no
+   * filename, but the function says the filename is "-". Avoid setting in this
+   * case; otherwise, all the evaluated/cli calls would match.
+   */
+  if ((NULL == wraprec->filename) && (NULL != filename)
```

Comment:
Should this also be checking if `wraprec` is NULL?

Reply:
Maybe. However this code is a verbatim copy of https://github.com/newrelic/newrelic-php-agent/blob/main/agent/php_user_instrument.c#L134-L151. The intention of this PR is not to change existing wraprec lookup implementations but to factor them out into separate modules for easier comparison testing, maintenance and review.

Reply:
Understood but if for robustness the value needs to be validated we need to make sure this update occurs in some fashion.

Reply:
I'm thinking that maybe in the final version of this PR, the *linked list* and *util_hashmap* lookup methods (and their implementations in `agent/php_user_instrument_llist.h` and `agent/php_user_instrument_util_hashmap.h`) should be removed completely. I left them in for now for following reasons:
1. In case someone would like to experiment with this codebase
2. In case when reviewing the code someone would like to quickly see how it was done before
If these reasons are no longer valid, I'll go ahead and remove these two implementations from the PR.